src/azul/service/responseobjects/storage_service.py

```diff
@@ -1,34 +1,38 @@
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dataclasses import dataclass
 from logging import getLogger
-from typing import Optional
+from typing import Optional, List
 import boto3
 from azul import config
 
 logger = getLogger(__name__)
+AWS_S3_DEFAULT_MINIMUM_PART_SIZE = 5242880  # 5 MB; see https://docs.aws.amazon.com/AmazonS3/latest/dev/qfacts.html
+MULTIPART_UPLOAD_MAX_WORKERS = 12
 
 
 class StorageService:
 
     def __init__(self):
-        self.__bucket_name = config.s3_bucket
-        self.__client = None  # the default client will be assigned later to allow patching.
+        self.bucket_name = config.s3_bucket
+        self._client = None  # the default client will be assigned later to allow patching.
 
     @property
```

Comment:
Check out `@lru_cache` from `functools`. It makes lazy init a one-liner (at a significant increase in spatial expense, but the concision and elimination of explicit state is worth it.